### PR TITLE
Don't use web middleware on delivery routes

### PR DESCRIPTION
**Pull request by holyfabi**
### **PR Type**
Enhancement

___

### **Description**
- Introduced a new middleware group for delivery-specific routes in `RouteServiceProvider`.
- Configured `SubstituteBindings` middleware for the new 'delivery' group in `bootstrap/app.php`.
- Created a new `routes/delivery.php` for handling delivery-specific routes, including an image delivery route.
- Removed the image route from `routes/web.php`, transferring responsibility to the new delivery routes.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>RouteServiceProvider.php</strong><dd><code>Introduce Delivery Middleware Route Group</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
app/Providers/RouteServiceProvider.php

- Added a new route group for 'delivery' middleware.

</details>
    

  </td>
  <td><a href="https://github.com/cybex-gmbh/transmorpher/pull/52/files#diff-846ccd1dd2844a1dab103a38b7919ae15e1d9cde8fae8507369b324ea1718689">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>delivery.php</strong><dd><code>Establish New Delivery Routes File</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
routes/delivery.php

<li>Created a new route file for delivery-specific routes.<br> <li> Defined a route for image delivery with specific transformations.<br>

</details>
    

  </td>
  <td><a href="https://github.com/cybex-gmbh/transmorpher/pull/52/files#diff-dea66756b2dba684d407f449d912c2610eca7c2e0e791b6056e6915023fbf846">+17/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>web.php</strong><dd><code>Remove Image Route from Web Routes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
routes/web.php

<li>Removed image route and related imports, which are now in delivery <br>routes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/cybex-gmbh/transmorpher/pull/52/files#diff-193e04aa1705f6f3e484d7efca8f45fe4d19d0ece3b6e6880d3ea070938a1fad">+0/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>app.php</strong><dd><code>Configure Middleware for Delivery Routes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
bootstrap/app.php

<li>Added <code>SubstituteBindings</code> middleware to the 'delivery' middleware <br>group.<br>

</details>
    

  </td>
  <td><a href="https://github.com/cybex-gmbh/transmorpher/pull/52/files#diff-10bc2462d34ebc59a5d482a1faca04ce74f6df89fcc2ef8bfb17b939647ea518">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: bootstrap/app.php [hunk 1]
```diff
@@ -1,17 +1,45 @@
...
+                ->group(base_path('routes/delivery.php'));
+        },
     )
     ->withMiddleware(function (Middleware $middleware) {
-        //
+        $middleware->appendToGroup('delivery', [
+            SubstituteBindings::class
+        ]);
+
+        $middleware->append([
```
Comment (gael-connan-cybex) [inline bootstrap/app.php:34]:
```suggestion
        // Add middlewares to all routes.
        $middleware->append([
```

File: bootstrap/app.php [hunk 2]
```diff
@@ -1,17 +1,45 @@
...
+            // \App\Http\Middleware\TrustHosts::class,
+            TrustProxies::class,
```
Comment (gael-connan-cybex) [inline bootstrap/app.php:36]: do we really need to add these here for them to be active? trimstrings sounds pretty basic

Comment (holyfabi) [reply bootstrap/app.php]: No, this were duplicates of the default global middlewares. You can check which global middleswares are applied with: `$middleware->getGlobalMiddleware()`, which includes all of the above listed middlewares.

**Review by gael-connan-cybex — APPROVED** [review]
tested on pullpreview. multiple images can now be uploaded when the client and the server run on the same domain.

**Pull request merged**